pkg/aiengine/data.go

```diff
@@ -55,26 +56,30 @@ func SendData(pod *pods.Pod, podState ...*state.State) error {
 
 		record := getProcessedRecord(pod, state)
 
+		serverMutex.Lock()
```

Comment:
Would this be better in the go func? If you defer the Unlock it should happen after the close.